### PR TITLE
Migrate Shipping block of new Order View page

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrates Shipping block to new Order View page.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #15824 
| How to test?  | Shipping block should behave the same as in legacy.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16065)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ *#}
+
+<div class="modal fade" id="updateOrderShippingModal" tabindex="-1" role="dialog">
+  {{ form_start(updateOrderShippingForm, {
+    'action': path('admin_orders_update_shipping', {'orderId': orderForViewing.id})
+  }) }}
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title">{{ 'Edit shipping details'|trans({}, 'Admin.Orderscustomers.Feature') }}</h5>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig:33]:
```suggestion
        <h5 class="modal-title">{{ 'Edit'|trans({}, 'Admin.Actions') }}</h5>
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/shipping.html.twig [hunk 2]
```diff
@@ -23,57 +23,63 @@
...
-        <td class="text-right">
-          {% if carrier.canEdit %}
-            <a href="#">
-              {{ 'Edit'|trans({}, 'Admin.Actions') }}
-            </a>
-          {% endif %}
-        </td>
+        <th>{{ 'Date'|trans({}, 'Admin.Global') }}</th>
+        <th>&nbsp;</th>
+        <th>{{ 'Carrier'|trans({}, 'Admin.Shipping.Feature') }}</th>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/shipping.html.twig:32]:
```suggestion
        <th>{{ 'Carrier'|trans({}, 'Admin.Global') }}</th>
```

**Comment by sarjon** [comment]
@LouiseBonnard both suggestions accepted, thanks! :)

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ *#}
+
+<div class="modal fade" id="updateOrderShippingModal" tabindex="-1" role="dialog">
+  {{ form_start(updateOrderShippingForm, {
+    'action': path('admin_orders_update_shipping', {'orderId': orderForViewing.id})
+  }) }}
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title">{{ 'Edit'|trans({}, 'Admin.Actions') }}</h5>
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig]: I'm surprised about this modal title, shouldn't it be something like `Edit Shipping` or `Update carrier` ?
@LouiseBonnard what do you think?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      <div class="modal-body">
+        {% if not 'PS_ORDER_RECALCULATE_SHIPPING'|configuration %}
+          <div class="alert alert-info" role="alert">
+            <p class="alert-text">
+              {{ 'Please note that carrier change will not recalculate your shipping costs, if you want to change this please visit Shop Parameters > Order Settings'|trans({}, 'Admin.Orderscustomers.Notification') }}
+            </p>
+          </div>
+        {% endif %}
+
+        <div class="form-group">
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig:47]: I guess you can't use `ps.form_group_row` inside a modal?

**Review by jolelievre — COMMENTED** [review]
Nice job @sarjon Just a couple questions to be sure and we're good

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig (hunk 3 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig]: It's 'Edit', it has been seen and validated with @MatShir yesterday. ;-)

**Comment by sarjon** [comment]
Rebased.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig (hunk 4 above)
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/Modal/update_shipping_modal.html.twig]: It has a little bit different layout than usual, so it must be rendered manually.

**Review by jolelievre — DISMISSED** [review]
Thanks @sarjon

**Comment by Robin-Fischer-PS** [comment]
Hi @sarjon !

I've two bugs on this PR :

- When trying to edit a shipping line, there is an error, and changes are not saved

- When editing a shipping line, the "Carrier" field should have the actual carrier selected by default. Instead, it's the first value in the dropdown that is selected (on the video, it should be "My carrier", but it's "Youpila" instead) : 

You can see both errors on this screen record : 
https://drive.google.com/file/d/1RaPssdHAn9rnapI2za0jLHl4lSp5i6BU/view?usp=sharing

Furthermore, I saw two bugs that were already present in Legacy. Would it be easy to fix for you ? (of course, it absolutely not mandatory, since these are no regression)

- When a product has a specific carrier selected (in product -> Shipping), on the shipping block, it should not be possible to choose another carrier (it's actually possible).


- When a carrier has max size limitation, on the shipping block, it should not be possible to choose this carrier when a product exceed this limit (it's actually possible).

If you need more info on these bugs, don't hesitate to ping me :smile: 


Thanks !

**Comment by sarjon** [comment]
Hi @Robin-Fischer-PS ,

> When trying to edit a shipping line, there is an error, and changes are not saved

I cannot reproduce it. :confused: Is there anything specific to this order? Maybe you need to rebuild assets?

> When editing a shipping line, the "Carrier" field should have the actual carrier selected by default. Instead, it's the first value in the dropdown that is selected (on the video, it should be "My carrier", but it's "Youpila" instead) :

Fixed.

> Furthermore, I saw two bugs that were already present in Legacy. Would it be easy to fix for you ?

I've checked those bugs, but it seems they are out of scope for this PR, I guess you could open an issue so they won't be forget. :)

**Comment by sarahdib** [comment]
Hello @sarjon 

Weird for the first issue I reproduce it. When I tried to edit the shipping line, there is an error, and changes are not saved

**Comment by matks** [comment]
@sarahdib Just checked, I have the error too, it is "An error occurred while sending an email to the customer." . It seems that because SMTP is not enabled in our test shops, we cannot modify shipping informations for an order 😅 as PS attempts to notify the user by mail.

**Comment by sarahdib** [comment]
@matks I configure SMTP and the problem is the same

**Comment by sarjon** [comment]
@sarahdib did you rebuild assets?

**Comment by matks** [comment]
@sarahdib You should have a more detailed error message in your logs (I guess you're in DEV environment, so this means the file var/logs/dev.log). It should be the line with a `php.CRITICAL: Uncaught Exception:`

**Comment by Robin-Fischer-PS** [comment]
Hi @sarjon ! Yes, we tried with and without rebuilding assets. @matks I'm looking at the log.

**Comment by matks** [comment]
It was a "carrier ID invalid" exception 🤔 I'm going to try to understand

**Comment by matks** [comment]
@sarjon Looks like `$updateOrderShippingForm` is initialized withouth a `current_order_carrier_id` parameter, consequently `UpdateOrderShippingDetailsHandler::handle()` crashes when calling

```
$orderCarrier = new OrderCarrier($command->getCurrentOrderCarrierId());
```
because `$command->getCurrentOrderCarrierId()` is `null`

**Comment by sarjon** [comment]
@matks does this happen for virtual order? Can you reproduce it on demo orders as well? If so, which one?

**Comment by matks** [comment]
I see it for demo orders with IDs 4 and 5 🤔

**Comment by sarjon** [comment]
> I see it for demo orders with IDs 4 and 5

That is so weird. I reinstalled PS, rebuilt assets (make sure hard refresh browser) and still cannot reproduce this issue. :scream_cat:

**Comment by matks** [comment]
Indeed after a full reinstall + assets rebuilt I dont see the issue anymore.
@Robin-Fischer-PS can you check with
- checkout the branch
- rebuild assets
- install the shop

**Comment by Robin-Fischer-PS** [comment]
Hi again @sarjon , @matks .

So it seems that this bug does not appear when trying with a dev core near me, I think it's some kind of voodoo bug.

Anyway, it now works fine, I don't know why.

I've detected a new bug : When editing a tracking number, it does not appear on the block (but it's OK on Legacy page, so the edit works fine).

See video : 
https://drive.google.com/file/d/1nbaM2aefZylc0WAvlt0ujTZY4PpQdOLp/view?usp=sharing

And @sarjon , it's been a pleasure to work with you, I wish you all the best for your next job :smile:

**Comment by matks** [comment]
@Robin-Fischer-PS OK, I checked the code. The tracking number was not displayed because no tracking URL was provided for the selected carrier.

I improved the behavior:
- when a tracking number is available, but no tracking URL, I show the number
- when both are available, I show the number + the link

**Review by matks — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @matks , thanks for the fix and the small improvement ! It's :heavy_check_mark: :)

**Comment by matks** [comment]
Thank you @sarjon for the PR and @Robin-Fischer-PS for the validation

**Pull request merged**